### PR TITLE
ids fix

### PR DESCRIPTION
**Pull request by thcr1**

### CODE REVIEW

File: Algorithms/Ids.py [hunk 1]
```diff
@@ -3,51 +3,56 @@
...
-        steps = run_dfs_in_layer(stack, visited, goal, layer, [])
 
-        if steps is not None or layer > max_depth:
+        steps = run_dfs_in_layer(board, layer)
+
+        if steps is not None:
             break
         layer += 1
 
     if not steps:
```
Comment (Stav-Elizur) [inline Algorithms/Ids.py]: You arrived this condition if the steps is not None according to the while loop

File: Algorithms/Ids.py [hunk 2]
```diff
@@ -3,51 +3,56 @@
...
+        if board.get_board_state() == board.get_goal_state():
+            return path
 
-        visited.add(node.__str__())
+        if depth == max_depth:
+            continue
 
-        # Get all available movements and reverse the order
-        new_boards = node.get_new_boards()[::-1]
+        if board in visited:
```
Comment (Stav-Elizur) [inline Algorithms/Ids.py]: Move this condition before stack.append

File: Algorithms/Ids.py [hunk 3]
```diff
@@ -3,51 +3,56 @@
...
+            continue
 
-        moves = []
+        visited.add(board)
 
-        for new_board_inx in range(0, len(new_boards)):
-            stack.append(new_boards[new_board_inx][0])
-            steps_to_goal = run_dfs_in_layer(stack, visited, goal, layer - 1, array + [new_boards[new_board_inx][1]])
+        row, col = board.find_zero_index()
+        if row is None:
```
Comment (Stav-Elizur) [inline Algorithms/Ids.py]: Is it necessary condition?

**Review by Stav-Elizur — CHANGES_REQUESTED** [review]

File: Algorithms/Ids.py (hunk 1 above)
Comment (thcr1) [reply Algorithms/Ids.py]: I think this is what you meant, but I am not sure

**Pull request merged**